### PR TITLE
AAE-10547: Throw exception when user group does not exist for Keycloak implementation

### PR DESCRIPTION
**Pull request by scaccia**

### CODE REVIEW

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java [hunk 1]
```diff
@@ -242,19 +242,16 @@ public List<SecurityResponseRepresentation> getApplicationPermissions(String app
...
     @Override
     public List<User> findUsersByGroupName(String groupName) {
-        List<User> users = new ArrayList<>();
-        if (!StringUtils.isEmpty(groupName)) {
-            Optional<Group> groupOptional = findGroupStrictlyEqualToGroupName(groupName);
-            groupOptional.ifPresent(group ->
-                    users.addAll(getUsersByGroupId(group.getId())));
-        }
-        return users;
+        if (StringUtils.isEmpty(groupName)) throw new IdentityInvalidGroupException(groupName);
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java]: it works for null, may be you could use spring assertMethods here?

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/test/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementServiceTest.java [hunk 2]
```diff
@@ -569,10 +571,11 @@ void should_returnUsers_when_searchingByGroupName() {
     }
 
     @Test
-    void should_returnEmptyUserList_when_groupNameNotFound() {
-        List<User> users = keycloakManagementService.findUsersByGroupName("fakeGroup");
-
-        assertThat(users).isEmpty();
+    void should_throwException_when_groupNameNotFound() {
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/test/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementServiceTest.java:573]: could you test also null, thanks

**Review by jesty — CHANGES_REQUESTED** [review]

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/test/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementServiceTest.java [hunk 3]
```diff
@@ -59,6 +60,7 @@
 @ExtendWith(MockitoExtension.class)
 class KeycloakManagementServiceTest {
 
+    public static final String FAKE_GROUP = "fakeGroup";
```
Comment (miguelruizdev) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/test/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementServiceTest.java]: let's be more accurate with the wording:
`public static final String NON_EXISTENT_GROUP = "non-existent-group";`

**Review by miguelruizdev — CHANGES_REQUESTED** [review]

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java (hunk 1 above)
Comment (scaccia) [reply activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java]: With spring Assert I can't throw a specific `IdentityInvalidGroupException` exception

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java [hunk 4]
```diff
@@ -242,19 +242,18 @@ public List<SecurityResponseRepresentation> getApplicationPermissions(String app
...
 
-    private Optional<Group> findGroupStrictlyEqualToGroupName(String groupName) {
-        return findGroups(groupName).stream()
-            .filter(group -> groupName.equals(group.getName()))
-            .findFirst();
+    private Group findGroupStrictlyEqualToGroupName(String groupName) {
+        return
+            Optional.ofNullable(groupName)
+                .map(g -> findGroups(g).stream())
+                .orElse(Stream.empty())
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java:254]: is it required?

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java [hunk 5]
```diff
@@ -242,19 +242,18 @@ public List<SecurityResponseRepresentation> getApplicationPermissions(String app
...
+        return getUsersByGroupId(groupFound.getId());
     }
 
-    private Optional<Group> findGroupStrictlyEqualToGroupName(String groupName) {
-        return findGroups(groupName).stream()
-            .filter(group -> groupName.equals(group.getName()))
-            .findFirst();
+    private Group findGroupStrictlyEqualToGroupName(String groupName) {
+        return
+            Optional.ofNullable(groupName)
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java:251]: what happens with empty string? I guess you should filter empty strings

**Review by jesty — CHANGES_REQUESTED** [review]

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java (hunk 4 above)
Comment (scaccia) [reply activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java:254]: Yes, It is the only way that I have found

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java (hunk 5 above)
Comment (scaccia) [reply activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java:251]: done

**Review by miguelruizdev — APPROVED** [review]

**Review by jesty — APPROVED** [review]
LGTM

**Pull request merged**